### PR TITLE
ci: bump cibuildwheel to 2.19.2

### PR DESCRIPTION
**Pull request by gnufede**
Updates CI Build Wheel to the latest release, as our CI is blocked by this issue related to the EOL of Centos 7: 
https://github.com/pypa/cibuildwheel/issues/1915

## Checklist

- [ ] The PR description includes an overview of the change
- [ ] The PR description articulates the motivation for the change
- [ ] The change includes tests OR the PR description describes a testing strategy
- [ ] The PR description notes risks associated with the change, if any
- [ ] Newly-added code is easy to change
- [ ] The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- [ ] The change includes or references documentation updates if necessary
- [ ] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [ ] Title is accurate
- [ ] All changes are related to the pull request's stated goal
- [ ] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [ ] Testing strategy adequately addresses listed risks
- [ ] Newly-added code is easy to change
- [ ] Release note makes sense to a user of the library
- [ ] If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [ ] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: .github/workflows/build_python_3.yml [hunk 1]
```diff
@@ -87,6 +87,7 @@ jobs:
               mv $w {dest_dir}
             done &&
             rm -rf ./tempwheelhouse
+          CIBW_ENVIRONMENT_MACOS: "MACOSX_DEPLOYMENT_TARGET=12"
```
Comment (sanchda) [inline .github/workflows/build_python_3.yml:90]: Do you know whether setting this closer to setup.py would help native tools pick up minimum required dependencies?  As far as I can tell, this parameter is inferred from the environment in a lot of cases.  If we were more strict, we may not have to set it here.

... it's also possible that we've just hit the limit of what we can accomplish with these legacy OSX targets and our only option is to limit what we support.  Any clue how old "12" is versus the old target (10.9)?

Comment (gnufede) [reply .github/workflows/build_python_3.yml:90]: Don't mind this PR, I didn't see yours merged 🙈

**Comment by gnufede** [comment]
Already fixed by #9697

File: .github/workflows/build_python_3.yml (hunk 1 above)
Comment (sanchda) [reply .github/workflows/build_python_3.yml:90]: Oh.  Well, I think we'll have to do a PR like this sometime anyway, so getting runtime versions sorted out is a good thing.  :)

**Pull request closed without merging**